### PR TITLE
Bug755174 Document using Lots to create investment capital gains transactions

### PR DESCRIPTION
**Pull request by goodvibes2**

### CODE REVIEW

**Comment by jralls** [comment]
If you're going to submit multiple PRs you need to use feature branches. Otherwise the older ones' commits get included in the newer ones. 

Only 8c1bad9 should be titled Bug 755174, the other two are housekeeping.

I don't have time to review this morning, will attend to that later.

**Comment by goodvibes2** [comment]
Sorry. Shall I redo?

**Comment by jralls** [comment]
Please fix the commit messages on a02724c and beccc62 and close the other two PRs. Force-push the result to Github. Meanwhile I'll review your changes in 8c1bad9.

File: guide/C/ch_invest.xml [hunk 1]
```diff
@@ -1605,7 +1605,7 @@ Income
             <row>
               <entry>Assets:Brokerage Account:Stock:AMZN</entry>
 
-              <entry>-100</entry>
+              <entry>&mdash;100</entry>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: No, use &amp;minus; for negative numbers. em-dashes are the really long ones sometimes used as comma substitutes&mdash;as in this example&mdash;in english sentences.

File: guide/C/ch_invest.xml [hunk 2]
```diff
@@ -1744,7 +1744,7 @@ Income
             <row>
               <entry>Assets:Brokerage Account:Stock:IBM</entry>
 
-              <entry>-51.7598</entry>
+              <entry>&mdash;51.7598</entry>
 
               <entry>90.00</entry>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Everything above this line should really be in a separate commit. It's more housekeeping and doesn't really have anything to do with the bug.

File: guide/C/ch_invest.xml [hunk 3]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+              or losses based on the selected <emphasis>Basis calculation method
+              </emphasis> report option (Average, FIFO or LIFO). I.e. it does
+              not use any of the lot information, just the security transaction
+              splits.
+            </para>
+          
+            <para>If you are selling <emphasis>all</emphasis> of your holding of
+              a particular security, it doesn't matter which cost method you
+              use as the total cost of the securities sold will be the same.
+              It is only when you have partially sold a security, that the
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Tense change, from present ("If you are selling") to past-perfect ("you have partially sold").

File: guide/C/ch_invest.xml [hunk 4]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+      <sect3 id="invest-sellLimitations">
+        <title>Limitations</title>
+        
+        <para>There are some limitations to using this feature.</para>
+        
+        <itemizedlist>
+          <listitem>
+            <para><application>&app;</application> has only 1 method of
+              automatically linking buy transactions to sell transactions -
+              First In/ First Out (FIFO). I.e. The oldest securities are always
```
Comment (jralls) [inline guide/C/ch_invest.xml]: You call this a "cost method" below, which is clearer than "method of automatically linking...".

File: guide/C/ch_invest.xml [hunk 5]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+              method is a report option, it is preferable that the same costing
+              method is used for all securities. Otherwise, it may be necessary
+              to run multiple Advanced Portfolio Reports, 1 for each costing
+              method and only including the securites using that costing method.
+            </para>
+            
+            <note>
+              <para>Provision has been made in <application>&app;</application>
+                for specifying costing method per security account but this is
+                yet to be implemented.
```
Comment (jralls) [inline guide/C/ch_invest.xml]: I think you're explaining the "scrub lots" function here, but you need to be clear about it; it appears the note applies to the Advance Portfolio Report.

File: guide/C/ch_invest.xml [hunk 6]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+                  lots to create the gain/loss transaction(s). The report should
+                  be correct when you use the LIFO report option.
+                </para>
+              </listitem>
+
+              <listitem>
+                <para> If you cost your sales using <emphasis>Average Costing
+                  </emphasis>, if you scrub, the gain will be calculated using 
+                  the FIFO Cost (as this is the only <application>&app;
+                  </application> method currently implemented), but the report
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Huh? "cost basis method currently implemented in GnuCash", not "Gnucash method currently implemented".

File: guide/C/ch_invest.xml [hunk 7]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+      <title>Automatic Calculation of Capital Gains or Losses Using Lots</title>
+       
+      <para><application>&app;</application> can automatically calculate and
+        create the capital gain or loss transactions for security sales. Lots
+        are used to link particular sell transaction splits to the buy 
+        transaction splits so that the actual cost of the sold securities can be
+        used to calculate the gain or loss.
+      </para>
+
+      <para>If you make an error, you can delete the lot(s) and/or delete the
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Note rather than para.

File: guide/C/ch_invest.xml [hunk 8]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+      
+      <para><application>&app;</application> can automatically create lots or a
+        user may create lots. <application>&app;</application> can automatically
+        link buy and sell transaction splits to lots or a user may do the 
+        linking.
+      </para>
+
+      <para>
+        <application>&app;</application> can automatically <emphasis>scrub
+        </emphasis> a security account. <application>&app;</application>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: "GnuCash can automatically..." 3 times, and the three uses seem somewhat redundant. Can you roll it up into a single paragraph?

File: guide/C/ch_invest.xml [hunk 9]
```diff
@@ -1817,6 +1817,1129 @@ Income
         </mediaobject>
       </screenshot>
     </sect2>
+    
+    <sect2 id="invest-sellLots">
+      <title>Automatic Calculation of Capital Gains or Losses Using Lots</title>
+       
+      <para><application>&app;</application> can automatically calculate and
```
Comment (jralls) [inline guide/C/ch_invest.xml]: I think you need to lead with an explanation of what a "lot" is, then that GnuCash has a built-in lot management facility that can be used to keep track of capital gains.

File: guide/C/ch_invest.xml [hunk 10]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+            <para>If you are selling <emphasis>all</emphasis> of your holding of
+              a particular security, it doesn't matter which cost method you
+              use as the total cost of the securities sold will be the same.
+              It is only when you have partially sold a security, that the 
+              costing method is important, so that the Advanced Portfolio Report
+              costs (and therefore gains or losses) are correct for the 
+              remainder of the security holding going forward. As the costing
+              method is a report option, it is preferable that the same costing
+              method is used for all securities. Otherwise, it may be necessary
+              to run multiple Advanced Portfolio Reports, 1 for each costing
```
Comment (jralls) [inline guide/C/ch_invest.xml]: "once" or "one time" instead of "1".

File: guide/C/ch_invest.xml [hunk 11]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+              method and only including the securites using that costing method.
+            </para>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: None of which has anything at all to do with Lots. Delete this item.

File: guide/C/ch_invest.xml [hunk 12]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+              </para>
+            </note>
+        
+            <para>Given that the Advanced Portfolio Report doesn't use any of
+              the lot information, if you want the reported costs and
+              gains/losses to be consistent with the transaction/lot data, it is
+              pointless (other than for the automatic creation of the gain/loss
+              transaction) linking sales with specific acquisitions unless you
+              always sell a particular security LIFO (Last In, First Out). I.e.:
+            </para>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: This doesn't make sense to me. "If you want the reported costs and gains/losses to be consistent with the transaction/lot data, it is pointless...linking sales with specific acquisitions unless you always sell.. LIFO". 

If you want the APR and your books to be consistent and gains in your books are calculated FIFO with the Lots function, shouldn't the APR also be told to use FIFO?

File: guide/C/ch_invest.xml [hunk 13]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+        loss. If the capital gain/loss is not correct, the Trial Balance
+         (<menuchoice>
+            <guimenu>Reports</guimenu>
+            <guisubmenu>Income &amp; Expense</guisubmenu>
+            <guimenuitem>Trial Balance</guimenuitem>
+          </menuchoice>) bottom line total debits will not balance to 
+          total credits.
+      </para>
+                    
+      <sect3 id="invest-sellLimitations">
```
Comment (jralls) [inline guide/C/ch_invest.xml]: I think you need to explain the various workings first, then the limitations. Otherwise the reader has no framework to understand the limitations.

File: guide/C/ch_invest.xml [hunk 14]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+            </para>
+            
+            <note>
+              <para>Provision has been made in <application>&app;</application>
+                for specifying costing method per security account but this is
+                yet to be implemented.
+              </para>
+            </note>
+        
+            <para>Given that the Advanced Portfolio Report doesn't use any of
```
Comment (jralls) [inline guide/C/ch_invest.xml]: This isn't so much of a limitation as it is part of the procedure.

File: guide/C/ch_invest.xml [hunk 15]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+
+              <caption><para>An image of the <emphasis>Lots in Account
+                </emphasis> window.</para>
+              </caption>
+            </mediaobject>
+          </screenshot>
+        </figure>
+        
+        <sect4>
+          <title>Screen Elements</title>
```
Comment (jralls) [inline guide/C/ch_invest.xml]: This belongs in Help, not Guide.

File: guide/C/ch_invest.xml [hunk 16]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+                    </emphasis> if there are no splits in the lot, otherwise the
+                    date of the earliest split in this lot.</para>
+                </listitem>
+
+                <listitem><para><guilabel>Closed</guilabel> &mdash; <emphasis>
+                  Open</emphasis> if the lot is not fully sold, otherwise the
+                  date of the last sell split.</para>
+
+                  <note>
+                    <para>Due to <ulink
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Let's not document bugs in the Guide.

File: guide/C/ch_invest.xml [hunk 17]
```diff
@@ -1817,6 +1817,1129 @@ Income
...
+      </sect3>
+      
+      <sect3 id="invest-sellManual">
+        <title>Manual Lot Creation and Linking</title>
+         
+        <para>Before using this feature ensure you are familiar with
+          <xref linkend="invest-sellLimitations" />
+        </para>
+         
+        <para>Here is an example of selling part of a security holding using
```
Comment (jralls) [inline guide/C/ch_invest.xml]: Explain what it is and why you'd use it before launching into an example. It would be better to include the discussion of why one would use this only in cases where one is calculating cap gains based on either LIFO or sale of designated lots (the same thing for securities as far as US personal tax law is concerned) as the scrub function takes care of FIFO and neither works for average-cost.

**Comment by jralls** [comment]
Chris, I made a bunch of in-line comments. An executive summary is that you need to re-think the organization of the section. There isn't an explanation of Lots or inventory basis accounting anywhere in the Guide, and without one I think that your instructions won't make much sense to most users. Following that it needs an explanation of why and when one would be interested in using the concept for their investments. Next explain in broad-brush the GnuCash lots facility where you can mention that it knows only about FIFO but that there are ways to work around that and the differences with the Advanced Portfolio Report. After that introductory material you can get into the specific use-cases. The actual mechanics of the Lots window belongs in Help, not in the Guide.

File: guide/C/ch_invest.xml (hunk 1 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Hi John, 
Thanks for reviewing my mods. Yes, I've several times wondered  if a minus sign is actually supposed to be there. I found some-one else had used &mdash; and thought that made it really clear. I'll have a look at how &minus; looks. There are no examples of that already in the help or guide.
How do you want me to handle all your requested changes? A commit for each one or only where you say they should be a separate commit? If multiple commits, do I use the commit message for this whole mod or a specific commit message for each?

File: guide/C/ch_invest.xml (hunk 11 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: This section is to do with the limitation of using lots and the Adv Port Rpt. This section points out that if you're selling all of your holding, this limitation does not apply.
Maybe it should be a tip or a note but I don't want to use too many of them.
I do think on review the 2nd and 3rd sentences should be a separate paragraph (actually a tip). This IS the guide.

File: guide/C/ch_invest.xml (hunk 13 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: I considered that but it is important they know the limitations before they jump in and create a mess. Would it be acceptable if I moved the limitations to after the workings then added a link to the limitations where they currently are?

File: guide/C/ch_invest.xml (hunk 14 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Which line no's are you referring to?

File: guide/C/ch_invest.xml (hunk 16 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Why not? It is what it is. This could save people hours investigating to see if they should raise a bug. I'll happily update the documentation if it's fixed. Would removing the word 'bug' be sufficient?

**Comment by goodvibes2** [comment]
John said:
Please fix the commit messages on  a02724c  and  beccc62  and close the other two PRs. Force-push the result to Github. 

Just in case you didn't get any notification of my force push: Above is completed.

File: guide/C/ch_invest.xml (hunk 16 above)
Comment (jralls) [inline guide/C/ch_invest.xml]: Because bugs are their own documentation and because the documentation doesn't get updated often enough.

File: guide/C/ch_invest.xml (hunk 11 above)
Comment (jralls) [inline guide/C/ch_invest.xml]: I think the section is a digression on the APR and inventory cost models. A simple note that if one wants the numbers in the APR to match the book one must use the same cost model in both places is all that's appropriate here.

File: guide/C/ch_invest.xml (hunk 1 above)
Comment (jralls) [inline guide/C/ch_invest.xml]: I think the best way would be to reset, fix everything up, and re-commit. If you reorganize like I suggest in the general comment then the changes will be a bit messy and there's no real need to preserve that for posterity.

File: guide/C/ch_invest.xml (hunk 13 above)
Comment (jralls) [inline guide/C/ch_invest.xml]: Put the *examples* at the end, after the limitations. There are a lot of concepts for the user to get his head around before he's ready for limitations or examples.

File: guide/C/ch_invest.xml (hunk 14 above)
Comment (jralls) [inline guide/C/ch_invest.xml]: This whole paragraph element and the following itemized list. The APR is a separate feature from Lots. But better to reorganize most of it away, move all of the explanation of inventory cost models to its own subsection near the top&mdash;or maybe somewhere else with a reference to here&mdash;and just have a short &lt;note&gt; that one should use the same model for both Lots and the APR if one wants the results to match.

File: guide/C/ch_invest.xml (hunk 16 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: As bug has been fixed (in 2.6.14), I've removed the note about the bug from ch_invest.xml.

File: guide/C/ch_invest.xml (hunk 2 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Understood.

File: guide/C/ch_invest.xml (hunk 3 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Fixed

File: guide/C/ch_invest.xml (hunk 4 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: I have changed this to:
GnuCash implements only the First In/ First Out (FIFO) cost method when automatically linking buy transactions to sell transactions.

I feel it is important to mention automatic linking because you can use lots to do LIFO but not without manually doing the linking. I haven't mentioned LIFO here as that is mentioned later. If you'd like me to remove the reference to automatic linking, I think I should mention LIFO here too.

File: guide/C/ch_invest.xml (hunk 5 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Removed Note.

File: guide/C/ch_invest.xml (hunk 6 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Fixed

File: guide/C/ch_invest.xml (hunk 7 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: fixed

File: guide/C/ch_invest.xml (hunk 8 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: fixed

File: guide/C/ch_invest.xml (hunk 9 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: done

File: guide/C/ch_invest.xml (hunk 11 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: done

File: guide/C/ch_invest.xml (hunk 1 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: I see now `&minus;` is just a std minus sign. I'd like to continue using mdash.

Comment (goodvibes2) [inline guide/C/ch_invest.xml]: I don't think use of - is a mistake, I just think &mdash; is clearer and therefore better for people who are learning.
As you don't agree, I have reverted to -.

File: guide/C/ch_invest.xml (hunk 13 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Done. Also changed Limitations to Considerations.

File: guide/C/ch_invest.xml (hunk 14 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Done

File: guide/C/ch_invest.xml (hunk 15 above)
Comment (goodvibes2) [inline guide/C/ch_invest.xml]: Removed from Guide. Will add to Help under another branch/PR.

**Comment by goodvibes2** [comment]
Closing now as I will do a new pull request.
(I also have to recreate by local & github gnucash-docs as I pulled origin/master into my maint again #@$!)
